GP11 : DSA-SHA224 DSA-SHA256 algorithms

cedric-chaumont-st-dev opened the pull request.

cedric-chaumont-st-dev commented:
For testing, use optee_test PR#26

jenswi-linaro commented on core/tee/tee_svc_cryp.c:
```diff
@@ -3420,7 +3422,12 @@ TEE_Result tee_svc_asymm_verify(uint32_t state,
+    	/*
+    	 * Depending on the DSA algorithm, the size of the hash to consider
+	 * could be less or equal than he whole size of the
+	 * hash algorithm involved (related to the subprime).
```
You mean the hash can be truncated when signed?

cedric-chaumont-st-dev replied:
Yes. For DSA-SH256, the hash value generated is coded on 256 bits, but the DSA  (encrypted hash value) could be either on 224bits or 256 bits (depending on the subprime).

jenswi-linaro replied:
Is the subprime something like the modulus in RSA, that is, defining the size of the result of the operation?

Please update the comment with that clarification, also a short note on why it does not affect the security of the other algorithms would nice.

pascal-brand-st-dev commented:
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

cedric-chaumont-st-dev commented:
Updated. Squashed. Verified.
+-----------------------------------------------------
37980 subtests of which 1 failed
687 test cases of which 1 failed
0 test case was skipped
TEE test application done!

NOTE: GP test 7671 failing (patch on going)

jenswi-linaro commented:
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

pascal-brand-st-dev commented:
Please rebase on master and add the tags

cedric-chaumont-st-dev commented:
Rebase. Update git comment.

pascal-brand-st commented:
Merged manually
